lib/pavilion/schedulers/plugins/slurm.py

```diff
@@ -312,71 +384,42 @@ def _get_config_elems(self):
         return elems, validators, defaults
 
     @classmethod
-    def parse_node_list(cls, node_list) -> NodeList:
+    def parse_node_list(cls, node_list, full=True) -> NodeList:
         """Convert a slurm format node list into a list of nodes, and throw
         errors that help the user identify their exact mistake."""
         if node_list is None or node_list == '':
             return NodeList([])
 
-        match = cls.NODE_LIST_RE.match(node_list)
-        if match is None:
-            node_part_re = re.compile(cls.NODE_SEQ_REGEX_STR + r'$')
-            # The following is required to handle foo[3,6-9].
-            prev = ""
-            for part in node_list.split(','):
-                # Logic used to recombined 'foo[3', '6-9]' after split.
-                if prev:
-                    part = prev + "," + part
-                    prev = ""
-                if '[' in part and ']' not in part:
-                    prev = part
-                    continue
-                if not node_part_re.match(part):
-                    raise ValueError(
-                        "Invalid Node List: '{}'. Syntax error in item '{}'. "
-                        "Node lists components be a hostname or hostname "
-                        "prefix followed by a range of node numbers. "
-                        "Ex: foo003,foo0[10-20],foo[103-104],foo[10,12-14],foo-m11-16"
-                        .format(node_list, part)
-                    )
-
-            # If all the parts matched, then it's an overall format issue.
-            raise ValueError("Invalid Node List: '{}' "
-                             "Good Example: foo003,foo0[10-20],"
-                             "foo[103-104], foo[10,12-14]")
+        # Split the nodelist on commas to seperate sections.
+        node_sections = node_list.replace(" ","").split(',')
+        allnodes = []
+        inner_split = False
+        inner_collect = []
+
+        # Collect sections so that each element is either a single node string
+        # or a node set descriptor.
+        for node_section in node_sections:
+            if inner_split:
+                inner_collect.append(node_section)
+                if "]" in node_section:
+                    inner_split = False
+                    allnodes.append(','.join(inner_collect))
+                    inner_collect = []
+            elif "[" not in node_section:
+                allnodes.append(node_section)
+            else:
+                if "]" in node_section:
+                    allnodes.append(node_section)
+                else:
+                    inner_collect.append(node_section)
+                    inner_split = True
```

Comment:
You're still doing what I advised against, though this time as a state machine.  I'm not going to waste too much time arguing about this, but I really think the character by character method is much more readable.  It also has the added benefit of being able to find and print errors about extra parens and bad white space in a way that can give the exact context where it occurred.